#!/bin/bash
#$ -cwd
#$ -S /bin/bash
#$ -N align1
#$ -l mem_free=4G
#$ -pe threaded 4
#$ -e error_$TASK_ID.txt
#$ -o stdout_$TASK_ID.txt

# - - - - - - - - - - - - - - - - - - - - -
# Load required modules
# - - - - - - - - - - - - - - - - - - - - -

module load star/2.4.5a

# - - - - - - - - - - - - - - - - - - - - -
# Specify paths/variables
# - - - - - - - - - - - - - - - - - - - - -

toolpath=/local/apps/star/2.4.5a/bin/Linux_x86_64
star_index_path=/ifs/home/ay1392/ROT3/star
inputpath=/ifs/data/proteomics/projects/Anna/Retrotransp-transcription/Breast/aluY/fastq/Normals
outputpath=/ifs/data/proteomics/projects/Anna/Retrotransp-transcription/Breast/aluY/align1/Normals

# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Variables for processing the data in parallel
# DO NOT CHANGE!
# This code is meant to assign one sample to the analysis
# by listing all of the files in a directory (inputDir)
# and then finding the Nth one in the list. N will be
# assigned by the $SGE_TASK_ID variable generated by
# the SGE cluster.
# Note: $SGE_TASK_ID will only be assigned if the flag
# -t is used and the job is submitted by 'qsub -t 1:N script.sh'
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

# Assign index variable
number=`expr $SGE_TASK_ID - 1`

# Generate ls of input files
files=(${inputpath}/*.r1.fastq)

# Get fastq file from index variable
sequence_read1=${files[number]}

# Get the name of current working file
BN=`basename $sequence_read1 .r1.fastq`

sequence_read2=${inputpath}/${BN}.r2.fastq

echo $BN
echo $sequence_read2
echo $sequence_read1

# Print diagnostics
echo "- - - - Diagnostics - - - - - - - - - - -"
echo "Number of slots: $NSLOTS"
echo "Number of hosts: $NHOSTS"
echo "Number in Queue: $QUEUE"
echo -e "OS type: $SGE_ARCH"
echo -e "Working on Task ID: "${SGE_TASK_ID}" "
echo -e "Working on file: "${files[number]}" "
echo -e "Output directory: "${outputDir}" \n"
echo "Run parameters:"
hostname -f
date
pwd
echo -e "- - - - - - - - - - - - - - - - - - -  \n"


$toolpath/STAR \
--genomeDir $star_index_path \
--readFilesIn $sequence_read1 $sequence_read2 \
--runThreadN 4 \
--outFilterMultimapScoreRange 1 \
--outFilterMultimapNmax 20 \
--outFilterMismatchNmax 10 \
--alignIntronMax 500000 \
--alignMatesGapMax 1000000 \
--sjdbScore 2 \
--alignSJDBoverhangMin 1 \
--genomeLoad NoSharedMemory \
--readFilesCommand cat \
--outFilterMatchNminOverLread 0.66 \
--outFilterScoreMinOverLread 0.66 \
--sjdbOverhang 100 \
--outSAMstrandField intronMotif \
--outSAMtype None \
--outSAMmode None  \
--outFileNamePrefix $outputpath${BN}
